src/entity/controllers/entityController.ts

```diff
@@ -0,0 +1,61 @@
+import { Logger } from '@map-colonies/js-logger';
+import { RequestHandler } from 'express';
+import httpStatus, { StatusCodes } from 'http-status-codes';
+import { injectable, inject } from 'tsyringe';
+import { Services } from '../../common/constants';
+import { Entity, UpdateEntity } from '../models/entity';
+import { EntityManager } from '../models/entityManager';
+import { HttpError } from '../../common/errors';
+import { EntityAlreadyExistsError, EntityNotFoundError } from '../models/errors';
+import { FileNotFoundError } from '../../file/models/errors';
+
+type PostEntityHandler = RequestHandler<{ fileId: string }, string, Entity>;
+type PostEntitiesHandler = RequestHandler<{ fileId: string }, string, Entity[]>;
+type PatchEntityHandler = RequestHandler<{ fileId: string; entityId: string }, string, UpdateEntity>;
+
+@injectable()
+export class EntityController {
+  public constructor(@inject(Services.LOGGER) private readonly logger: Logger, private readonly manager: EntityManager) {}
+
+  public postEntity: PostEntityHandler = async (req, res, next) => {
+    try {
+      await this.manager.createEntity({ ...req.body, fileId: req.params.fileId });
+      return res.status(httpStatus.CREATED).send(httpStatus.getStatusText(httpStatus.CREATED));
+    } catch (error) {
+      if (error instanceof EntityAlreadyExistsError) {
+        (error as HttpError).status = StatusCodes.CONFLICT;
+      } else if (error instanceof FileNotFoundError) {
+        (error as HttpError).status = StatusCodes.NOT_FOUND;
+      }
+      next(error);
+    }
+  };
+
+  public postEntities: PostEntitiesHandler = async (req, res, next) => {
+    try {
+      const fileId = req.params.fileId;
+      const bodyWithFileId = req.body.map((entity) => ({ ...entity, fileId }));
```

Comment:
I do not think you should manipulate the request body, the signature of createEntities should be `createEntities(fileId, req.body)`

Reply:
done